### ISSUE TITLE
How to disable autorun in embeds?

### ISSUE THREAD

**Issue by moonpiedumplings** [live-codes/livecodes#344]
I am trying to embed this into my blog, to create a python tutorial for students, and I would it to not run things like `input()` on the load of the page, without doing lazy loading for each embed. I'm guessing it's the `autoupdate` configuration, but I can't figure out how to insert it into an embed. 

I'm trying to do something like this, but it isn't changing anything.

```{=html}
<div id="livecodes-qlbo5jm6qm">
  <span>Open the project <a href="https://livecodes.io/?x=id/tse96fdpgz4" target="_blank">Untitled Project</a> in <a href="https://livecodes.io/" target="_blank">LiveCodes</a></span>
</div>

<script src="https://livecodes.io/sdk/livecodes.umd.js"></script>
<script>
  const options = {
    "config": {
      "autoupdate": false,
      "tools": {
        "status": "open"
      }
    },
    "import": "id/tse96fdpgz4",
    "loading": "eager"
  };
  livecodes.createPlayground("#livecodes-qlbo5jm6qm", options);
</script>

```

In addition to this, is there a way to have the console be the default page open, rather than results?

**Comment by hatemhosny** [comment]
hi @moonpiedumplings ,

your imported project (id/tse96fdpgz4) already provides the configuration object, so your config settings are overridden.
To override project config, use [params](https://livecodes.io/docs/sdk/js-ts#params)

```html
<div id="livecodes-qlbo5jm6qm">
  <span>Open the project <a href="https://livecodes.io/?x=id/tse96fdpgz4" target="_blank">Untitled Project</a> in <a href="https://livecodes.io/" target="_blank">LiveCodes</a></span>
</div>

<script src="https://livecodes.io/sdk/livecodes.umd.js"></script>
<script>
const options = {
  params: {
    languages: "pyodide",
    autoupdate: false,
    preview: false,
    console: "full",
  },
  import: "id/tse96fdpgz4",
  loading: "click",
};
livecodes.createPlayground("#livecodes-qlbo5jm6qm", options);
</script>

```

The result always runs when the playground is first loaded. `autoupdate` determines what happens when the code changes. If you do not want to run the playground, you may use the embed option `loading: "click"`.

The `params.preview` option allows showing a faded preview of the html page behind the "click to load" screen. However, the script will still run before loading the playground. To prevent that, set it to false.

`console: "full"` is what you are asking for.

**Comment by hatemhosny** [comment]
I probably need to add a config option `autorun` to determine if the result page should run when the playground first loads.

also, the `preview` option needs to be documented

**Comment by hatemhosny** [comment]
> I probably need to add a config option `autorun` to determine if the result page should run when the playground first loads.
> 

on a second thought, if `autoupdate` is set to `false` the result page should not run, even when playground first loads.  
this will prevent the confusion of having 2 similar options with different meanings.

to handle the scenario of wanting to run the page once when the playground loads:

```js
import { createPlayground } from 'livecodes';

const options = {
  params: { autoupdate: false },
  // other embed options
};
createPlayground("#container", options).then(playground => {
  playground.run();
});

```

this is being addressed in #345 and should be available in next release

@moonpiedumplings thank you for reporting

**Comment by hatemhosny** [comment]
> The `params.preview` option allows showing a faded preview of the html page behind the "click to load" screen. However, the script will still run before loading the playground. To prevent that, set it to false.
> 

it is probably also not a good idea to have the preview page run by default, specially if the page sends alerts or starts sounds, etc.

**Comment by hatemhosny** [comment]
this is now fixed by #345 and should be available in next release  
you may test it here: https://dev.livecodes.io

this code should now work as expected:

```html
<div id="livecodes-qlbo5jm6qm">
  <span>Open the project <a href="https://livecodes.io/?x=id/tse96fdpgz4" target="_blank">Untitled Project</a> in <a href="https://livecodes.io/" target="_blank">LiveCodes</a></span>
</div>

<script src="https://livecodes.io/sdk/livecodes.umd.js"></script>
<script>
const options = {
  appUrl: 'https://dev.livecodes.io/',
  params: {
    languages: "pyodide",
    autoupdate: false,
    console: "full",
  },
  import: "id/tse96fdpgz4",
  loading: "eager",
};
livecodes.createPlayground("#livecodes-qlbo5jm6qm", options);
</script>
```
[open in LiveCodes](https://dev.livecodes.io/?x=id/8s96m6pxnts)

@moonpiedumplings please let me know if that solves your problem.

**Comment by hatemhosny** [comment]
@moonpiedumplings I will close this as fixed.

If you still have a problem with this issue please feel free to re-open it.
